pkg/handlers/config.go

```diff
@@ -64,6 +65,86 @@ type CurrentAppConfigResponse struct {
 	ConfigGroups []kotsv1beta1.ConfigGroup `json:"configGroups"`
 }
 
+type DownloadAppConfigResponse struct {
+	Success bool   `json:"success"`
+	Error   string `json:"error,omitempty"`
+}
+
+func (h *Handler) DownloadAppConfig(w http.ResponseWriter, r *http.Request) {
+	downloadAppConfigResponse := DownloadAppConfigResponse{
+		Success: false,
+	}
+
+	foundApp, err := store.GetStore().GetAppFromSlug(mux.Vars(r)["appSlug"])
+	if err != nil {
+		logger.Error(err)
+		downloadAppConfigResponse.Error = "failed to get app from app slug"
+		JSON(w, http.StatusInternalServerError, downloadAppConfigResponse)
+		return
+	}
+
+	sequence, err := strconv.Atoi(mux.Vars(r)["sequence"])
+	if err != nil {
+		logger.Error(err)
+		downloadAppConfigResponse.Error = "failed to parse app sequence"
+		JSON(w, http.StatusInternalServerError, downloadAppConfigResponse)
+		return
+	}
+
+	filename := mux.Vars(r)["filename"]
+	if filename == "" {
+		logger.Error(err)
+		downloadAppConfigResponse.Error = "failed to parse filename, parameter was empty"
+		JSON(w, http.StatusInternalServerError, downloadAppConfigResponse)
+		return
+	}
+
+	archiveDir, err := ioutil.TempDir("", "kotsadmconfig")
+	if err != nil {
+		logger.Error(err)
+		downloadAppConfigResponse.Error = "failed to create temp directory"
+		JSON(w, http.StatusInternalServerError, downloadAppConfigResponse)
+	}
+	defer os.RemoveAll(archiveDir)
+
+	configValue, err := getAppConfigValue(foundApp, int64(sequence), filename, archiveDir)
+	if err != nil {
+		logger.Error(err)
+		downloadAppConfigResponse.Error = "failed to get app config"
+		JSON(w, http.StatusInternalServerError, downloadAppConfigResponse)
+		return
+	}
+
+	decoded, err := base64.StdEncoding.DecodeString(configValue)
+	if err != nil {
+		logger.Error(err)
+		downloadAppConfigResponse.Error = "failed to decode config value"
+		JSON(w, http.StatusInternalServerError, downloadAppConfigResponse)
+	}
+
+	// create file in temp directory to use for io.Copy
+	err = os.WriteFile(fmt.Sprintf("%s/%s", archiveDir, filename), decoded, 0666)
+	if err != nil {
+		logger.Error(err)
+		downloadAppConfigResponse.Error = "failed to write temp config file"
+		JSON(w, http.StatusInternalServerError, downloadAppConfigResponse)
+	}
+
+	fileForDownload, err := os.Open(fmt.Sprintf("%s/%s", archiveDir, filename))
+	if err != nil {
+		logger.Error(err)
+		downloadAppConfigResponse.Error = "failed to open temp config file"
+		JSON(w, http.StatusInternalServerError, downloadAppConfigResponse)
+	}
+
+	fileStat, _ := fileForDownload.Stat()
```

Comment:
this error should be handled.  `fileStat` will be nil  on error and possibly cause a panic.